AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestRunnerServiceImpl.kt

```diff
@@ -213,6 +213,21 @@ internal class TestRunnerServiceImpl internal constructor(
                 }
             }
         }
+        // remove this if block when b\299975596 is fixed
+        if (mergedXmlBlobs.isEmpty()) {
+            if (byFullDeviceId.isEmpty()) {
+                return emptyList()
+            }
+            val deviceId = byFullDeviceId.values.first().deviceRun.deviceId
+            val mergedResults = byFullDeviceId.values.first().xmlResults.first()
```

Comment:
this is not merged anymore so we shouldn't call it `mergedResults`.

why do we return the first one instead of all available?

Reply:
Yeah, these are not mergedResults. But there should only be 1 result present in the byFullDeviceId map. So, using the xmlResult file here from that result would essentially have the same content as merged results file.

Reply:
then we shouldn't call the variable mergedResults.

Furthermore, i think it is more consistent for this to be:
if merged results exists, return it
otherwise, return all others.

`byDeviceId.values.map { TestRunResult(...) }`

Reply:
Made `mergedResults` nullable so that when mreged_results.xml file is not present, we can set it to null and incorporated rest of your feedback. Please review.